dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -801,6 +805,33 @@ private boolean restore(PageStoreDir pageStoreDir) {
     return true;
   }
 
+  private void addPageBasedOnPredicate(PageStoreDir pageStoreDir, PageInfo pageInfo) {
+    boolean tested = false;
+    if (mPagePredicate.isPresent()) {
+      tested = mPagePredicate.get().test(pageInfo);
+    }
+    if (!tested) {
+      addPageToDir(pageStoreDir, pageInfo);
+      MetricsSystem.histogram(MetricKey.CLIENT_CACHE_PAGES_AGES.getName())
+                   .update(System.currentTimeMillis() - pageInfo.getCreatedTimestamp());
+    }
+    else {
+      // delete page directly and no metadata put into meta store
+      boolean isPageDeleted = deletePage(pageInfo, false);
```

Comment:
@apc999 do we need to acquire lock when deleting? the signature said so but doesn't seems like we acquire lock when calling it

Reply:
we should . or we will have race condition.
we should fix other references too.

Reply:
We need to acquire lock when calling the `deletePage` method.